File: zantt-api-backend/src/main/java/com/zantt/entities/WorkspaceEntity.java
package com.zantt.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "workspaces")
public class WorkspaceEntity {
    @Id
    @Column(name = "workspace_id")
    private String workspaceId;
    @Column(name = "task_id")
    private String taskId;
    @Column(name = "project_Id")
    private String projectId;
    @Column(name = "title")
    private String title;
    @Column(name = "contents")
    private String contents;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getWorkspaceId() {
        return workspaceId;
    }

    public void setWorkspaceId(String workspaceId) {
        this.workspaceId = workspaceId;
    }

    public String getTaskId() {
        return taskId;
    }

    public void setTaskId(String taskId) {
        this.taskId = taskId;
    }

    public String getProjectId() {
        return projectId;
    }

    public void setProjectId(String projectId) {
        this.projectId = projectId;
    }

    public String getContents() {
        return contents;
    }

    public void setContents(String contents) {
        this.contents = contents;
    }
}

File: zantt-api-backend-net/Zantt/Entities/WorkspaceEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Zantt.Entities;

[Table("workspaces")]
public class WorkspaceEntity
{
    [Required]
    [Column("project_id")]
    [StringLength(255)]
    public string ProjectId { get; set; } = Guid.NewGuid().ToString();

    [Required]
    [Column("task_id")]
    [StringLength(255)]
    public string TaskId { get; set; } = Guid.NewGuid().ToString();

    [Key]
    [Column("workspace_id")]
    [StringLength(255)]
    public string WorkspaceId { get; set; } = Guid.NewGuid().ToString();

    [Required]
    [Column("title")]
    [StringLength(255)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [Column("contents")]
    [StringLength(2000)]
    public string Contents { get; set; } = string.Empty;

    [Required]
    [Column("created_time")]
    public DateTime CreatedTime { get; set; } = DateTime.Now;
}
